From fbebf8cddb59091644f9c2841899faad3ddcef4a Mon Sep 17 00:00:00 2001
From: Sujeet Baranwal <sbaranwal@marvell.com>
Date: Tue, 4 May 2021 14:34:59 -0700
Subject: [PATCH 1517/1921] dts: marvell: add CM3 SRAM memory to cp115 ethernet
 device tree

CM3 SRAM address space would be used for Flow Control configuration.

Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Signed-off-by: Sujeet Baranwal <sbaranwal@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/3419
Reviewed-by: Marcin Wojtas <Marcin.Wojtas@cavium.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Tested-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Change-Id: I0208cad9d0571c3814f49fa2f3a48996a2fbfc88
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/51108
Reviewed-by: Jia Ma <Jia.Ma@cavium.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-cp11x.dtsi | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
index 9b83409be816..96084bf63758 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
@@ -70,6 +70,7 @@
 			marvell,system-controller = <&CP11X_LABEL(syscon0)>;
 			status = "disabled";
 			dma-coherent;
+			cm3-mem = <&CP11X_LABEL(cm3_sram)>;
 
 			CP11X_LABEL(eth0): eth0 {
 				interrupts = <39 IRQ_TYPE_LEVEL_HIGH>,
@@ -220,6 +221,14 @@
 			};
 		};
 
+		CP11X_LABEL(cm3_sram): cm3@220000 {
+			compatible = "mmio-sram";
+			reg = <0x220000 0x800>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0 0x220000 0x800>;
+		};
+
 		CP11X_LABEL(rtc): rtc@284000 {
 			compatible = "marvell,armada-8k-rtc";
 			reg = <0x284000 0x20>, <0x284080 0x24>;
-- 
2.31.1

